DataFormats/Common/interface/DetSetVectorNew.h

```diff
@@ -464,27 +638,32 @@ namespace edmNew {
   }
 
   template<typename T>
-  inline void DetSetVector<T>::updateImpl(Item & item)  {
+  inline void DetSetVector<T>::updateImpl(Item & item) {
     // no getter or already updated
-    if (getter.empty()) assert(item.offset>=0);
-    if (item.offset!=-1 || getter.empty() ) return;
-    item.offset = int(m_data.size());
-    FastFiller ff(*this,item,true);
-    (*boost::any_cast<std::shared_ptr<Getter> >(&getter))->fill(ff);
+    if (getter.empty()) { assert(item.isValid()); return;}
+#ifdef USE_ATOMIC
```

Comment:
Is this conditional still needed since you nicely hid the use of atomic within ```item```?